discovery-provider/src/utils/ipfs_lib.py

```diff
@@ -40,32 +40,33 @@ def get_metadata(self, multihash, metadata_format, user_replica_set=None):
         """
         logger.warning(f"IPFSCLIENT | get_metadata - {multihash}")
         api_metadata = metadata_format
-        retrieved_from_local_node = False
-        retrieved_from_gateway = False
+        retrieved = False
         start_time = time.time()
 
-        # First try to retrieve from local ipfs node.
-        try:
-            api_metadata = self.get_metadata_from_ipfs_node(multihash, metadata_format)
-            retrieved_from_local_node = api_metadata != metadata_format
-        except Exception:
-            logger.error(
-                f"Failed to retrieve CID from local node, {multihash}", exc_info=True
-            )
-
-        # Else, try to retrieve from gateways.
-        if not retrieved_from_local_node:
-            try:
-                api_metadata = self.get_metadata_from_gateway(
-                    multihash, metadata_format, user_replica_set
-                )
-                retrieved_from_gateway = api_metadata != metadata_format
-            except Exception:
-                logger.error(
-                    f"Failed to retrieve CID from gateway, {multihash}", exc_info=True
-                )
+        with concurrent.futures.ThreadPoolExecutor(max_workers=2) as executor:
+            metadata_futures = []
+            metadata_futures.append(executor.submit(
+                self.get_metadata_from_ipfs_node, multihash, metadata_format
+            ))
+            metadata_futures.append(executor.submit(
+                self.get_metadata_from_gateway, multihash, metadata_format, user_replica_set
+            ))
+            for get_metadata_future in concurrent.futures.as_completed(metadata_futures):
+                original_task = metadata_futures[get_metadata_future]
+                print(f'IPFSCLIENT | original task: {original_task.task}')
+                try:
+                    api_metadata = get_metadata_future.result()
+                    retrieved = api_metadata != metadata_format
```

Comment:
this isn't something you introduced in this PR, but this naming convention here somewhat confuses me? if `api_metadata` is the ipfs response, and the `metadata_format` is the "required format for user md", wouldn't we want to use `==`  to indicate that the DP successfully retrieved md? 

I looked at the code, and to my understanding, it looks that if fetching the data from ipfs or the gateway returns data with missing fields that those fields get defaulted to `None`. but wouldn't this comparison always evaluate to false then? 

cc @SidSethi / @hareeshnagaraj  on this comparison. maybe im misinterpreting some logic here

Reply:
seems like metadata_format is an unclear name. maybe we can call it like `default_metadata_fields` and add comments for this arg

as far as the `==` i think `!=` is actually correct since retrieved is checking that what you get from the request is not the same as the default